pkgs/by-name/co/colstr/package.nix

```
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitea
+, nix-update-script
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "colstr";
+  version = "1.0.0";
+
+  src = fetchFromGitea {
+    domain = "git.sleeping.town";
+    owner = "wonder";
+    repo = "colstr";
+    rev = finalAttrs.version;
+    hash = "sha256-0V2S/yYu5L7qxkT4Zf18x9+cHoPMztFmgSywpxF8QqA=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    eval ${stdenv.cc.targetPrefix}c++ main.cpp
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 a.out $out/bin/colstr
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    description = "Deterministically output each input argument in a color assigned to it";
+    homepage = "https://git.sleeping.town/wonder/colstr";
+    license = with licenses; [ cc0 ];
+    maintainers = with maintainers; [ annaaurora ];
+    mainProgram = "colstr";
```

```suggestion
    mainProgram = "colstr";
    platforms = platforms.unix;
```

It's not so nice to have `10.rebuild-darwin: 0` when the package builds fine there.

---

how do we know it only works on unix?

---

It could of course be `platforms.all` or `platforms.unix ++ platforms.windows`, the only important thing being that it's defined and covers four Hydra platforms.

---

oh, i might've been used to language specific derivations setting it for me